### PR TITLE
Cypress ctm200 command injection

### PR DESCRIPTION
**Pull request by 0x45dd**
Architecture Support: Added support for ARM-based payloads alongside the existing command-based payloads, including the use of the CmdStager mixin for ARM targets.
TARGETURI Handling: Updated the TARGETURI option to handle non-standard installation directories by defaulting to / and adjusted the request path accordingly.
Option Cleanup: Removed explicit datastore options (RHOST, LHOST, LPORT) as they are included with the payload and no longer needed in the module's options.

### CODE REVIEW

**Comment by bwatters-r7** [comment]
It looks like this is failing our liniting process:
```
Rubocop failed. Please run rubocop -a modules/exploits/linux/http/cypress_ctm200_command_injection.rb and verify all issues are resolved
```

**Comment by wvu** [comment]
Deleted a bunch of files?

**Comment by bwatters-r7** [comment]
This is the second PR I've seen files deleted in that I think was unintended....?
https://github.com/rapid7/metasploit-framework/pull/19368

File: modules/exploits/linux/http/cypress_ctm200_command_injection.rb [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+        'Privileged'     => true,
+        'Payload'        => {
+          'Space'       => 2048,
+          'BadChars'    => ''
+        },
+        'Platform'       => ['unix'],
+        'Arch'           => [ARCH_CMD, ARCH_ARM],
+        'Targets'        => [
+          ['Automatic', {}],
+          ['ARM Linux', { 'Arch' => ARCH_ARM }]
```
Comment (bwatters-r7) [inline modules/exploits/linux/http/cypress_ctm200_command_injection.rb:44]:
```suggestion
          ['ARM Linux', { 'Arch' => ARCH_ARM }],
          ['CMD Linux', { 'Arch' => ARCH_CMD }]
```

File: modules/exploits/linux/http/cypress_ctm200_command_injection.rb [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+      return CheckCode::Vulnerable
+    end
+
+    CheckCode::Safe
+  end
+
+  def exploit
+    if target.name == 'ARM Linux'
+      # Use CmdStager for ARM payloads
+      cmd_exec payload.encoded
```
Comment (bwatters-r7) [inline modules/exploits/linux/http/cypress_ctm200_command_injection.rb:100]: I don't see where `cmd_exec` defined for this context?
To use CmdStager, you must implement the `execute_command` method, then call the builtin `execute_cmdstager` method that will use your `execute_command` method.  See https://docs.metasploit.com/docs/development/developing-modules/guides/how-to-use-command-stagers.htm

**Comment by bwatters-r7** [comment]
> Deleted a bunch of files?

I think the author's antivirus ate them.

**Comment by smcintyre-r7** [comment]
I'm going to attic this. If you'd like to continue working on it, just let us know and we can reopen it. Thanks!

**Pull request closed without merging**